modules/native_streaming_client_module/src/native_device_impl.cpp

```diff
@@ -482,33 +482,21 @@ NativeDeviceImpl::~NativeDeviceImpl()
     }
 }
 
-void NativeDeviceImpl::initStatuses()
-{
-    if (!this->context.getTypeManager().hasType("ConnectionStatusType"))
-    {
-        const auto statusType = EnumerationType("ConnectionStatusType", List<IString>("Connected",
-                                                                                      "Reconnecting",
-                                                                                      "Unrecoverable"));
-        this->context.getTypeManager().addType(statusType);
-    }
-    const auto statusInitValue = Enumeration("ConnectionStatusType", "Connected", this->context.getTypeManager());
-    this->statusContainer.asPtr<IComponentStatusContainerPrivate>().addStatus("ConnectionStatus", statusInitValue);
-}
-
 // INativeDevicePrivate
-void NativeDeviceImpl::publishConnectionStatus(ConstCharPtr statusValue)
+void NativeDeviceImpl::publishConnectionStatus(const EnumerationPtr& status)
 {
-    auto newStatusValue = this->statusContainer.getStatus("ConnectionStatus");
-    newStatusValue = statusValue;
-
-    this->statusContainer.asPtr<IComponentStatusContainerPrivate>().setStatus("ConnectionStatus", newStatusValue);
```

Comment:
Should we maybe still keep this for backwards compatibility purposes? @dejancrn what are your thoughts on this?

Reply:
Restored for backward compatibility